dev/com.ibm.ws.kernel.feature.core/src/com/ibm/ws/kernel/feature/resolver/FeatureResolver.java

```diff
@@ -233,15 +233,15 @@ public Result resolveFeatures(Repository repository, Collection<ProvisioningFeat
     /**
      * Resolves a collection of root features against a repository.
      *
-     * @param repository            the feature repository to use
-     * @param kernelFeatures        the set of kernel features to use for auto-feature processing
-     * @param rootFeatures          the root features to resolve
-     * @param preResolved           the set of already resolved features to base the resolution delta off of
-     * @param allowMultipleVersions a flag that allows multiple versions (this flag will effectively ignore singletons)
-     * @param supportedProcessTypes the supported process types to allow to be resolved
+     * @param repository              the feature repository to use
+     * @param kernelFeatures          the set of kernel features to use for auto-feature processing
+     * @param rootFeatures            the root features to resolve
+     * @param preResolved             the set of already resolved features to base the resolution delta off of
+     * @param allowedMultipleVersions the set that includes features that effectively ignore singletons (null means none, empty set means all)
+     * @param supportedProcessTypes   the supported process types to allow to be resolved
      * @return the resolution result
      */
     public Result resolveFeatures(Repository repository, Collection<ProvisioningFeatureDefinition> kernelFeatures, Collection<String> rootFeatures, Set<String> preResolved,
-                                  boolean allowMultipleVersions,
+                                  Set<String> allowedMultipleVersions,
```

Comment:
The feature resolver is used outside of liberty (WDT, migration tools, maybe install).  I worry this is a breaking change instead of adding a new method.

Reply:
I thought of that as well.  I can make a new method instead.

Reply:
PR is updated to have a new method.